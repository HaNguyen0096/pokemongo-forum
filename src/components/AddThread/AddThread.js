import React, { Component } from 'react'
import TopicContext from '../../contexts/TopicContext'
import ApiService from '../../services/api-service'
import { Button, Textarea } from '../Utils/Utils'
import './AddThread.css'

export default class AddThread extends Component {
  
  static contextType = TopicContext

  handleSubmit = ev => {
    ev.preventDefault()
    const { topic } = this.context
    console.log(topic)
    const { title, content } = ev.target

    ApiService.postThread(topic.id, title.value, content.value)
      .then(this.context.addThread)
      .catch(this.context.setError)
  }

  render() {
    return (
      <form
        className='Thread'
        onSubmit={this.handleSubmit}
      >
        <div className='addThread'>
          <div className='title'>
          <Textarea
            required
            aria-label='Thread Title...'
            name='title'
            id='title'
            cols='30'
            rows='2'
            placeholder='Thread Title..'>
          </Textarea>
          </div>
          <div className='content'>
          <Textarea
            required
            aria-label='Type your thread...'
            name='content'
            id='content'
            cols='30'
            rows='4'
            placeholder='Type your thread..'>
          </Textarea>
          </div>
        </div>

        <Button type='submit'>
          Post Thread
        </Button>
      </form>
    )
  }
}
