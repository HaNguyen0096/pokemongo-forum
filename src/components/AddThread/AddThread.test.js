import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddThread from './AddThread'

describe('AddThread component', () => {

  it('renders AddThread without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <AddThread />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for AddThread as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <AddThread />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders AddThread by default', () => {
    const wrapper = shallow(<AddThread  />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})