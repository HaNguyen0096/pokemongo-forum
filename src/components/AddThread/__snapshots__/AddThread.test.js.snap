// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddThread component renders AddThread by default 1`] = `
<form
  className="Thread"
  onSubmit={[Function]}
>
  <div
    className="addThread"
  >
    <div
      className="addTitle"
    >
      <Textarea
        aria-label="Thread Title..."
        cols="30"
        id="title"
        name="title"
        placeholder="Thread Title.."
        required={true}
        rows="2"
      />
    </div>
    <div
      className="addContent"
    >
      <Textarea
        aria-label="Type your thread..."
        cols="30"
        id="content"
        name="content"
        placeholder="Type your thread.."
        required={true}
        rows="4"
      />
    </div>
  </div>
  <Button
    className="postThreadBtn"
    type="submit"
  >
    Post Thread
  </Button>
</form>
`;

exports[`AddThread component renders the UI for AddThread as expected 1`] = `
<form
  className="Thread"
  onSubmit={[Function]}
>
  <div
    className="addThread"
  >
    <div
      className="addTitle"
    >
      <textarea
        aria-label="Thread Title..."
        className="Textarea "
        cols="30"
        id="title"
        name="title"
        placeholder="Thread Title.."
        required={true}
        rows="2"
      />
    </div>
    <div
      className="addContent"
    >
      <textarea
        aria-label="Type your thread..."
        className="Textarea "
        cols="30"
        id="content"
        name="content"
        placeholder="Type your thread.."
        required={true}
        rows="4"
      />
    </div>
  </div>
  <button
    className="Button postThreadBtn"
    type="submit"
  >
    Post Thread
  </button>
</form>
`;
